src/shadow.tsx

```diff
@@ -39,33 +51,21 @@ export function makeShadowRaw(rootEl: Element, childNodes?: Iterable<Node>, { mo
 //     return newroot
 // }
 
-function putDomIntoShadow(shadow_root: GShadowRoot, childNodes: Iterable<Node>) {
-    const fragment = document.createDocumentFragment()
-    for (const node of childNodes) {
-        fragment.appendChild(node)
+function putDomIntoShadow(shadowRoot: GShadowRoot, childNodes: Iterable<Node>) {
+    if (childNodes) {
+        shadowRoot.append(...Array.from(childNodes));
     }
-    shadow_root.appendChild(fragment)
 }
 
-const virtual_root = document.createDocumentFragment()
-
 export const ShadowStyle = defineComponent({
     props: {
         media: String,
         nonce: String,
     },
     setup(props, { slots }) {
-        return (): VNode => (
-            <style media={props.media} nonce={props.nonce}>
-                {slots.default?.()}
-            </style>
-        )
+        return () => h('style', { media: props.media, nonce: props.nonce }, slots.default?.());
```

Comment:
JSX is used here intentionally, do not modify

Reply:
I would like to know in more detail why you use JSX?